### ISSUE TITLE
Issue with seeding monai transforms in dataloader workers

### ISSUE THREAD

**Issue by nkaenzig** [kaiko-ai/eva#843]
MONAI's random transforms use numpy.random for random number generation
which is seeded at the very beginning by lightning's seed_everything, but when
torch spins up the dataloader workers, it will only reseed torch's random states
and not numpy - so you basically end up with multiple dataloader workers that
have equally seeded random transforms, resulting in redundant transform outputs
and therefore reducing the diversity of the resulting training data.
To solve this, this method should be called in the dataloader's worker_init_fn
with a unique seed for each worker.
